File: apps/ios/Pebbles/Features/Path/Models/Visibility.swift
import Foundation

enum Visibility: String, CaseIterable, Identifiable, Hashable, Decodable {
    // M51 privacy grades: secret = owner-only (the default since the backfill),
    // private = owner + mutual connections, public = anyone with the /p link.
    case secret = "secret"
    case `private` = "private"
    case `public` = "public"

    var id: String { rawValue }

    var label: LocalizedStringResource {
        switch self {
        case .secret:  return "Secret"
        case .private: return "Connections"
        case .public:  return "Public"
        }
    }

    /// SF Symbol per grade — mirrored by the Android badge/menu drawables.
    var systemImageName: String {
        switch self {
        case .secret:  return "lock.fill"
        case .private: return "person.2.fill"
        case .public:  return "globe"
        }
    }
}

File: apps/android/app/src/main/kotlin/app/pbbls/android/features/path/models/Visibility.kt
package app.pbbls.android.features.path.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Pebble privacy grade (M51) — mirrors iOS `Visibility.swift`. `@SerialName`
 * on each entry decodes/encodes the DB strings, so a payload's
 * `visibility: Visibility` field serializes straight to the wire string.
 *
 * `SECRET` = owner-only (the default since the M51 backfill), `PRIVATE` =
 * owner + mutual connections, `PUBLIC` = anyone with the /p link.
 */
@Serializable
enum class Visibility {
    @SerialName("secret")
    SECRET,

    @SerialName("private")
    PRIVATE,

    @SerialName("public")
    PUBLIC,
}
